{
	"id": "CVE-2019-20916",
	"vulnerabilities": [
		{
			"content": {
				"id": "CVE-2019-20916",
				"title": "Unaffected components for: python-pip: directory traversal in _download_http_url() function in src/pip/_internal/download.py (moderate)",
				"description": "Red Hat's versions of the associated software have been determined to NOT be affected by CVE-2019-20916.",
				"severity": [
					{
						"type": "vendor",
						"source": "secalert@redhat.com",
						"vendor": "moderate"
					},
					{
						"type": "cvss_v31",
						"source": "secalert@redhat.com",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:H/I:H/A:H",
							"base_score": 8,
							"base_severity": "HIGH",
							"temporal_score": 8,
							"temporal_severity": "HIGH",
							"environmental_score": 8,
							"environmental_severity": "HIGH"
						}
					}
				],
				"cwe": [
					{
						"source": "secalert@redhat.com",
						"cwe": [
							"CWE-22"
						]
					}
				],
				"references": [
					{
						"source": "secalert@redhat.com",
						"url": "https://access.redhat.com/security/cve/CVE-2019-20916"
					}
				],
				"published": "2019-04-16T00:00:00Z",
				"modified": "2023-12-15T00:00:00Z"
			}
		},
		{
			"content": {
				"id": "CVE-2019-20916",
				"title": "python-pip: directory traversal in _download_http_url() function in src/pip/_internal/download.py (moderate)",
				"description": "DOCUMENTATION: A flaw was found in the pip package installer for Python when downloading or installing a remote package via a specified URL. Improper validation of the \"Content-Disposition\" HTTP response header makes a path traversal attack possible, leading to an arbitrary file overwrite. This flaw allows an attacker who controls a malicious server to execute arbitrary code on the system. \n            STATEMENT: This issue has been rated as having Moderate impact because of the preconditions needed to trigger the flaw: it only affects Python Wheels and requires the user to pip-install a wheel from a malicious server. Installing software from untrusted servers is insecure by definition and strongly discouraged, as it may lead to system compromise regardless of this CVE.\n\nThis flaw did not affect the versions of `python-pip` in Python 3.8 as shipped with Red Hat Enterprise Linux 8 and Red Hat Software Collections 3, as they already included the fix for this CVE.\n            MITIGATION: Avoid downloading or installing packages from potentially malicious servers via the command-line \"pip download\" or \"pip install\".",
				"severity": [
					{
						"type": "vendor",
						"source": "secalert@redhat.com",
						"vendor": "moderate"
					},
					{
						"type": "cvss_v31",
						"source": "secalert@redhat.com",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:H/I:H/A:H",
							"base_score": 8,
							"base_severity": "HIGH",
							"temporal_score": 8,
							"temporal_severity": "HIGH",
							"environmental_score": 8,
							"environmental_severity": "HIGH"
						}
					}
				],
				"cwe": [
					{
						"source": "secalert@redhat.com",
						"cwe": [
							"CWE-22"
						]
					}
				],
				"references": [
					{
						"source": "secalert@redhat.com",
						"url": "https://access.redhat.com/security/cve/CVE-2019-20916"
					}
				],
				"published": "2019-04-16T00:00:00Z",
				"modified": "2023-12-15T00:00:00Z"
			},
			"segments": [
				{
					"ecosystem": "redhat:7",
					"tag": "rhel-7-including-unpatched"
				}
			]
		}
	],
	"detections": [
		{
			"ecosystem": "redhat:7",
			"conditions": [
				{
					"criteria": {
						"operator": "OR",
						"criterias": [
							{
								"operator": "AND",
								"criterions": [
									{
										"type": "version",
										"version": {
											"vulnerable": true,
											"fix_status": {
												"class": "unfixed",
												"vendor": "Will not fix"
											},
											"package": {
												"name": "python-pip"
											}
										}
									}
								]
							},
							{
								"operator": "AND",
								"criterions": [
									{
										"type": "version",
										"version": {
											"vulnerable": true,
											"fix_status": {
												"class": "unfixed",
												"vendor": "Will not fix"
											},
											"package": {
												"name": "python3-pip"
											}
										}
									}
								]
							}
						]
					},
					"tag": "rhel-7-including-unpatched"
				}
			]
		}
	],
	"data_source": {
		"id": "redhat-ovalv2",
		"raws": [
			"vuls-data-raw-redhat-oval-repository-to-cpe/repository-to-cpe.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/definitions/oval:com.redhat.cve:def:201920916.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/objects/rpminfo_object/oval:com.redhat.cve:obj:201820225001.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/objects/rpminfo_object/oval:com.redhat.cve:obj:201820225002.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/tests/rpminfo_test/oval:com.redhat.cve:tst:201820225001.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/tests/rpminfo_test/oval:com.redhat.cve:tst:201820225002.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/tests/rpminfo_test/oval:com.redhat.cve:tst:201820225003.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/tests/rpminfo_test/oval:com.redhat.cve:tst:201820225004.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/tests/rpmverifyfile_test/oval:com.redhat.cve:tst:20042779005.json",
			"vuls-data-raw-redhat-ovalv2/7/rhel-7-including-unpatched/tests/rpmverifyfile_test/oval:com.redhat.cve:tst:20042779006.json",
			"vuls-data-raw-redhat-ovalv2/8/rhel-8-including-unpatched/definitions/oval:com.redhat.unaffected:def:201920916.json"
		]
	}
}
